Wikipedia/View Controllers/WebView/WebViewController.m

```diff
@@ -51,6 +57,11 @@ @interface WebViewController () <LanguageSelectionDelegate, FetchFinishedDelegat
 
 @property (strong, nonatomic) WMFArticlePopupTransition* popupTransition;
 
+@property (nonatomic, strong) NSMutableArray* nativeTitleLabelModelsArray;
```

Comment:
Instead of the array… we can add a "SectionTitlesViewController" that manages the array of models / views that float on the web view. So that logic is then encapsulated separately.

You can also move the 3 properties below to that class.